### ISSUE TITLE
 PlusServerOpenIGTLinkCommandsTest	fails on Linux due to timeout

### ISSUE THREAD

**Issue by Sunderlandkyl** [PlusToolkit/PlusLib#246]
http://perkdata.cs.queensu.ca/CDash/testDetails.php?test=1547459&build=82204

The test passes, but it takes ~377s, way beyond the current 90s timeout.

@lassoan: Is this a problem that needs to be fixed, or should the timeout threshold just be increased?

**Comment by Sunderlandkyl** [comment]
In general, all of the commands seem to take longer.
In particular, the SaveConfig command seems to take a very long time.
There are several others that take a while as well though.

Could this be due to the fact that the server is running in the VirtualBox?

**Comment by adamrankin** [comment]
Yes, is this still running on Tesla? The virtual machine only has 1 core, which will considerably affect performance.

**Comment by adamrankin** [comment]
That being said

http://perkdata.cs.queensu.ca/CDash/viewTest.php?onlyfailed&buildid=82346

is a native Ubuntu 4 core (8 HT) with 16GB of RAM, and it still times out.

**Comment by Sunderlandkyl** [comment]
The tests are running on Artery, although it's using the same virtual machine from Tesla (copied and renamed to artery-virtualBox).

**Comment by Sunderlandkyl** [comment]
We found out yesterday that the test hangs on Linux waiting for mutex locks in vtkPlusOpenIGTLinkClient::SendCommand() and vtkPlusOpenIGTLinkClient::SendMessage().

This only happens when the PlusServer process is launched from PlusServerRemoteControl, but not when it's started separately.

For now, the conclusion is that we will disable the test on Linux, and potentially try to fix this at a later time.

**Comment by adamrankin** [comment]
Thanks for investigating, why disable the test though? It's telling us useful information.

**Comment by lassoan** [comment]
We track this error using the issue tracker. Having constantly failing tests would make it more difficult to detect actual errors/regressions.

**Comment by adamrankin** [comment]
Wait, I thought this is an actual error?

**Comment by adamrankin** [comment]
Is PlusServerRemoteControl the small network app responsible for launching plus servers? If so, is that a direction we were looking to make the "normal" process?

**Comment by lassoan** [comment]
> Is PlusServerRemoteControl the small network app responsible for launching plus servers?

Yes, it is only a test application. One of its features is that it can start a PlusServer process (again, just for testing). PlusServerRemoteControl does not have a network interface.

PlusServerLauncher will have the ability to launch PlusServer processes based on requests that it receives through OpenIGTLink.

**Comment by adamrankin** [comment]
Ah ok, cool!
